bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/CloseEditorHandler.java

```diff
@@ -33,43 +33,42 @@
  *
  * @since 3.3
  */
-public class CloseEditorHandler extends AbstractEvaluationHandler {
-
-	private Expression enabledWhen;
-
-	public CloseEditorHandler() {
-		registerEnablement();
-	}
+public class CloseEditorHandler extends AbstractHandler {
 
 	@Override
-	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IWorkbenchWindow window = HandlerUtil.getActiveWorkbenchWindowChecked(event);
-		IEditorPart part = HandlerUtil.getActiveEditorChecked(event);
-		window.getActivePage().closeEditor(part, true);
+	public Object execute(ExecutionEvent event) {
```

Comment:
IMO, we should always try to look in the event to ensure we get the right editor. Other strategies can have flaws and return another editor than the one that triggered the event.
Would it be possible to get the e4 element (MPart) from the event context instead?

If not, I think it would be worth keeping usage of `HandlerUtil.getActiveEditorChecked(event)` if it's not null, and if null, then failback to an alternative strategy.

Reply:
Thanks for the feedback - yes that makes sense, I have updated the PR now such that the event context is used to retrieve the active part.  If the active part is an IEditorPart I perform the previous logic, else if it is an E4Wrapper such as when we contribute a PartDescriptor via a model fragment it is dealt with via the EPartService.